Spigot-API-Patches/0191-Add-PlayerSettingsChangeEvent.patch

```diff
@@ -0,0 +1,121 @@
+From e2a607d5a57cb63975f022bc1b67af07781f1c8b Mon Sep 17 00:00:00 2001
+From: MiniDigger <admin@minidigger.me>
+Date: Mon, 20 Jan 2020 21:38:34 +0100
+Subject: [PATCH] Add PlayerSettingsChangeEvent
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/event/player/PlayerSettingsChangeEvent.java b/src/main/java/com/destroystokyo/paper/event/player/PlayerSettingsChangeEvent.java
+new file mode 100644
+index 00000000..2d27b6d8
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/player/PlayerSettingsChangeEvent.java
+@@ -0,0 +1,106 @@
++package com.destroystokyo.paper.event.player;
++
++import org.jetbrains.annotations.NotNull;
++
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++import org.bukkit.inventory.MainHand;
++
++/**
++ * Called when the player changes his settings
++ */
++public class PlayerSettingsChangeEvent extends PlayerEvent {
++
++    private static final HandlerList handlers = new HandlerList();
++
++    private final String locale;
++    private final int viewDistance;
++    private final ChatVisibility chatVisibility;
++    private final boolean chatColors;
++    private final int skinparts;
++    private final MainHand mainHand;
++
++    public PlayerSettingsChangeEvent(@NotNull Player player, @NotNull String locale, int viewDistance, @NotNull ChatVisibility chatVisibility, boolean chatColors, int skinParts, @NotNull MainHand mainHand) {
++        super(player);
++        this.locale = locale;
++        this.viewDistance = viewDistance;
++        this.chatVisibility = chatVisibility;
++        this.chatColors = chatColors;
++        this.skinparts = skinParts;
++        this.mainHand = mainHand;
++    }
++
++    @NotNull
++    public String getLocale() {
++        return locale;
++    }
++
++    public int getViewDistance() {
++        return viewDistance;
++    }
++
++    @NotNull
++    public ChatVisibility getChatVisibility() {
++        return chatVisibility;
++    }
++
++    public boolean hasChatColorsEnabled() {
++        return chatColors;
++    }
++
++    public int getSkinparts() {
++        return skinparts;
++    }
++
++    @NotNull
++    public MainHand getMainHand() {
++        return mainHand;
++    }
++
++    public boolean hasCapeEnabled() {
```

Comment:
let's not make the impl detail of skins api

Reply:
Mmh, ok, I will let the event return a new SkinParts API class with an impl that handles that.

Reply:
done